pkg/scheduler/scheduler.go

```diff
@@ -223,7 +226,11 @@ func (k *KubernetesScheduler) StatusUpdate(driver bindings.SchedulerDriver, task
 		}
 		fallthrough
 	case mesos.TaskState_TASK_KILLED:
-		k.taskRegistry.UpdateStatus(taskStatus)
+		_, state := k.taskRegistry.UpdateStatus(taskStatus)
+		//TODO(jdef) what if I receive this after a TASK_LOST or TASK_KILLED? I don't want to reincarnate then..
```

Comment:
It's possible that master may report TASK_LOST and subsequently report a non-terminal status. TASK_LOST is special this way, unlike the other terminal statuses. We already track *finished* tasks (for some reason) -- we could just as easily track *lost* tasks and handle this TASK_LOST-followed-by-TASK_STARTING more gracefully.